java/debugger.jpda/src/org/netbeans/modules/debugger/jpda/models/ShortenedStrings.java

```diff
@@ -385,19 +349,220 @@ else if (valuesField.type() instanceof ArrayType &&
+     * @param sourceArray Backing array reference. May be a byte or char array
+     * @param backing
+     * @param isLittleEndian
+     * @param start
+     * @param length
+     * @param dest
+     */
+    private static void copyToCharArray(List<Value> sourceArray, int srcPos,
+            char[] dest, int destPos, int length, InternalStringEncoding backing,
+            boolean isLittleEndian) throws IOException{
+        if (backing == InternalStringEncoding.CHAR_ARRAY){
+            //that was easy
+            for (int i = 0; i < length; i++) {
+                Value v = sourceArray.get(i + srcPos);
+                if (!(v instanceof CharValue)){
+                    throw new IOException(MessageFormat.format("Char at {0} "
+                            + "is not a character: {1}", srcPos + i, v));
+                }
+                dest[destPos + i] = ((CharValue)v).charValue();
+            }
+            return;
+        }
+        if (backing == InternalStringEncoding.BYTE_ARRAY_LATIN1){
+            //that was also easy
+            for (int i = 0; i < length; i++) {
+                Value v = sourceArray.get(i + srcPos);
+                if (!(v instanceof ByteValue)){
+                    throw new IOException(MessageFormat.format("Char at {0} "
+                            + "is not a byte: {1}", srcPos + i, v));
+                }
+                char c = (char)((ByteValue)v).byteValue();
+                //strip off the sign value
+                c &= 0xff;
+                dest[destPos + i] = c;
+            }
+            return;
+        }
+        //uft16 it is
+        assert backing == InternalStringEncoding.BYTE_ARRAY_UTF16;
+        for (int i = 0; i < length; i++) {
+            Value left = sourceArray.get(i * 2);
+            Value right = sourceArray.get((i * 2) + 1);
+            if (!(left instanceof ByteValue && right instanceof ByteValue)){
+                throw new IOException(MessageFormat.format("Char at {0} is "
+                        + "not a byte pair: {1},{2}", srcPos + i, left, right));
+            }
+            dest[destPos + i] = utf16Combine(((ByteValue)left).byteValue(),
+                    ((ByteValue)right).byteValue(), isLittleEndian);
+        }
+    }
+    private static char utf16Combine(byte left, byte right, boolean isLittleEndian){
+        int hiByteShift, lowByteShift;
+        if (isLittleEndian){
+            hiByteShift = 0;
+            lowByteShift = 8;
+        }
+        else{
+            hiByteShift = 8;
+            lowByteShift = 0;
+        }
+        char c1 = (char)left;
+        char c2 = (char)right;
+        //remove the extended sign
+        c1 = (char) (0xFF & c1);
+        c2 = (char) (0xFF & c2);
+        char c = (char)(c1 << hiByteShift |
+                c2 << lowByteShift);
+        return c;
+    }
+    private static int length(int arrayLength, InternalStringEncoding backingEncoding){
```

Comment:
Please also make sure, that there is an empty line between the end of the previous method and the next one.